### PR TITLE
Create test from index

### PR DESCRIPTION
**Pull request by RealJoelRivera**
Created a function called `createTestFromIndex` that sets `testsDiff` to the new value in state for `tests` .  

Added the data only from the chosen index position specified by the user to `testsDiffClone` . 

`setTestsDiff` with the specified data added to `testsDiffClone` .

### CODE REVIEW

**Review by conor-jh — COMMENTED** [review]
I think this is a great first pass at the functionality—and I like the implementation of defining an index to create new tests from. A couple things to consider:

- The index input could be improved—we can probably just use a modified version of the `InlineInput` component. But also, it’d be great to restrict the input to numbers between 0–n (where n is the max. index in the payload)
- Similarly, do we _want_ users to be able to enter an invalid index? Right now, it fails gracefully and just produces a empty object but there are valid arguments for throwing a more explicit error. 
- Could we do away with the secondary button entirely? That is, if an the index input is filled, create from index, otherwise, just copy the whole object?

In short—I’m down to approve this for functionality, which is solid. But there are definitely some UX things to discuss… I’m happy to talk ’em over as a group and pair program if that’s helpful. Let me know 👌

**Review by Donaldblanc — APPROVED** [review]
ok The core functionality works. We will need to rethink how you are pulling out the test instead of reaching out to the DOM each time for performance.  Each test is stored in state and we should figure out some logic to pull it from there. Button will need some logic and UI work but it is ok for this PR.  Thanks for working on this.

**Review by conor-jh — APPROVED** [review]
In the future, let’s to to separate linting changes from functionality but this looks good! We can clean up the UI a bit in the coming days.
